src/transforms/groupby.js

```diff
@@ -35,6 +35,35 @@ exports.attributes = {
             'with `x` [1, 3] and one trace with `x` [2, 4].'
         ].join(' ')
     },
+    nameformat: {
+        valType: 'string',
+        description: [
+            'Pattern by which grouped traces are named. If only one trace is present,',
+            'defaults to the group name (`"%{group}"`), otherwise defaults to the group name',
+            'with trace name (`"%{group} (%{trace})"`). Available escape sequences are `%{group}`, which',
+            'inserts the group name, and `%{trace}`, which inserts the trace name. If grouping',
+            'GDP data by country when more than one trace is present, for example, the',
+            'default "%{group} (%{trace})" would return "Monaco (GDP per capita)".'
+        ].join(' ')
+    },
+    groupnames: {
```

Comment:
Hmm. I missed this during my first review. I'm not a fan of `groupnames` being an array container. 

We could instead declare `groupnames` as a `data_array` and expecting its length to be equal to `groups.length`, but that would lead to some (maybe a lot of) duplication. 

Maybe we could `groupname` attribute to the `styles` container below instead which already expects a target group?

Actually, do we even need a `groupnames` attribute? `groups` items can be any strings, so setting `groupnames[i].name` is equivalent to changing the corresponding `groups` item.

I apologize for not catching this sooner.